pkg/server/test/connected_objects.go

```diff
@@ -1222,8 +1255,10 @@ func ConnectedObjectsTest(t *testing.T, ds storage.OpenFGADatastore) {
 			timeoutCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 			defer cancel()
 
+			resolutionMetadata := connectedobjects.NewResolutionMetadata()
+
 			go func() {
-				err = connectedObjectsCmd.Execute(timeoutCtx, test.request, resultChan)
+				err = connectedObjectsCmd.Execute(timeoutCtx, test.request, resultChan, resolutionMetadata)
```

Comment:
this change here is odd.. i think it'd be better if you called `resolutionMetadata := connectedobjects.NewResolutionMetadata()` inside of `Execute` and then change the return value of `Execute` to return the resolution metadata

Reply:
The problem with creating NewResolutionMetadata inside connected objects query's Execute function instead of passing in as parameter is that connectedObjectsCmd.Execute is used as go func in ListObjectsQuery's [evaluate function](https://github.com/openfga/openfga/blob/e7dc8c6e06946b6d8b98676b5f338add29f8dcf9/pkg/server/commands/list_objects.go#L229).  This means that we will either 

- have to create a separate channel in https://github.com/openfga/openfga/blob/e7dc8c6e06946b6d8b98676b5f338add29f8dcf9/pkg/server/commands/list_objects.go#L228, listen on the channel and sum all the values at the end  OR
- wait for connectedObjectsQuery.Execute to return all the resolution metadata value before [processing connectedObjectsResChan](https://github.com/openfga/openfga/blob/e7dc8c6e06946b6d8b98676b5f338add29f8dcf9/pkg/server/commands/list_objects.go#L254).

Both options are not ideal.

Reply:
Ah.... `evaluate` fires the call but doesn't wait for a response, because StreamedListObjects wouldn't work properly if we did wait.